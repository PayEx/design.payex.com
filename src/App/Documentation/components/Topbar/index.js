import React, { Component } from "react";
import PrismCode from "react-prism";

import { ComponentPreview, DocContainer, Property, Attribute } from "#";
import TopbarComponent from "@/Topbar";
import { topbar } from "$/px-script/main";

const menu = {
    btn: {
        icon: "menu",
        text: "Menu"
    },
    items: ["Link 1", "Link 2", "Link 3"]
};

const Overview = () => (
    <>
        <h2 id="overview">Overview</h2>
        <p>
            To use a topbar the bare minimum markup required is:
        </p>
        <ul>
            <li><Property value=".topbar" /> wrapper.</li>
            <li>Anchor with class <Property value=".topbar-logo" /> to display the logo and make it clickable.</li>
        </ul>
        <p>
            With this you get a topbar with the PayEx logo.
            In the topbar you can add additional functionality like a menu containing links. To enable this add a <PrismCode className="language-html">{"<button>"}</PrismCode> and
            a <PrismCode className="language-html">{"<nav>"}</PrismCode> containing anchors. Remember to include <Attribute data name="toggle-nav" value="{your_nav_id}"/> as a <PrismCode className="language-html">{"<button>"}</PrismCode> attribute
            to get the menu to appear when the menu-button is clicked.
        </p>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <TopbarComponent topbarContent={menu} logout />
        </ComponentPreview>
    </>
);

class Topbar extends Component {
    componentDidMount () {
        topbar.init(1);
    }

    render () {
        return (
            <DocContainer docToc>
                <p className="lead">The topbar is used to give users an easily available navigational bar at the top of your web application.</p>
                <Overview />
            </DocContainer>
        );
    }
}

export default Topbar;

/* For testing */
export { Overview };
