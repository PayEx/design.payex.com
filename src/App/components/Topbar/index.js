import React, { Fragment } from "react";
import PropTypes from "prop-types";
import classnames from "classnames";

const TopbarBtn = ({ icon, text }) => {
    const iconClass = classnames("material-icons", icon === "menu" ? "topbar-menu-icon" : "");

    if (icon || text) {
        return (
            <button type="button" className="topbar-menu">{"\n\t\t"}
                    <>
                        <i className={iconClass}>{icon}</i>
                        {"\n\t\t"}
                    </>
            </button>
        );
    }

    return null;
};

const TopbarMenu = ({ menu, logout }) => {
    const { items } = menu;

    return (
        <nav className="topbar-nav">
            {items.map((item, i) => (
                <Fragment key={i}>
                    {"\n"}
                    <a href="#" className={item.name === "Home" ? "active" : null} onClick={e => e.preventDefault()}>
                        <i className="material-icons">{item.icon}</i>
                        <span>{item.name}</span>
                    </a>
                </Fragment>
            ))}{"\n"}
            {logout ?
                <>
                    <a className="topbar-link-right" href="#" onClick={e => e.preventDefault()}>{"\n"}Log out {"\n"}<i className="material-icons">exit_to_app</i>{"\n"}</a>{"\n"}
                </> : null}
        </nav>
    );
};

const Topbar = ({ topbarContent, wide, logout }) => (
    <header className={`topbar${wide ? " topbar-wide" : ""}`}>{"\n"}
        {topbarContent ?
        <>
            <TopbarBtn icon={topbarContent.btn.icon} text={topbarContent.btn.name} target={topbarContent.id} />{"\n"}
            {"\n"}<a href="#" className="topbar-logo" onClick={e => e.preventDefault()}></a>{"\n"}
            <TopbarMenu menu={topbarContent} logout={!!logout} />
        </> :
        <>
            {"\n"} <a href="#" className="topbar-logo" onClick={e => e.preventDefault()}></a>{"\n"}
        </>}
    </header>
);

Topbar.propTypes = {
    topbarContent: PropTypes.object,
    fixed: PropTypes.bool,
    logout: PropTypes.bool
};

export default Topbar;
