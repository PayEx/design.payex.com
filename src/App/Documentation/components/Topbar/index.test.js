import React from "react";
import { shallow } from "enzyme";

import Topbar, { Overview, FixedTopbar } from "./index";

describe("Components: Topbar", () => {
    it("is defined", () => {
        expect(Topbar).toBeDefined();
    });

    it("renders", () => {
        const wrapper = shallow(<Topbar />);

        expect(wrapper).toMatchSnapshot();
    });

    describe("Overview", () => {
        it("is defined", () => {
            expect(Overview).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<Overview />);

            expect(wrapper).toMatchSnapshot();
        });
    });
});
