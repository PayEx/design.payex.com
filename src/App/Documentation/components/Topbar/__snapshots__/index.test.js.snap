// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components: Topbar FixedTopbar renders 1`] = `
<Fragment>
  <h2
    id="fixed-topbar"
  >
    Fixed topbar
  </h2>
  <p>
    Add 
    <Property
      value=".topbar-fixed"
    />
     with your 
    <Property
      value=".topbar"
    />
     to get a fixed topbar. For a demonstration of the usage see the documentations top bar or implement it in your own project!
  </p>
  <ComponentPreview
    codeFigure={true}
    language="html"
  >
    <Topbar
      fixed={true}
    />
  </ComponentPreview>
</Fragment>
`;

exports[`Components: Topbar Overview renders 1`] = `
<Fragment>
  <h2
    id="overview"
  >
    Overview
  </h2>
  <p>
    To use a topbar the bare minimum markup required is:
  </p>
  <ul>
    <li>
      <Property
        value=".topbar"
      />
       wrapper.
    </li>
    <li>
      Anchor with class 
      <Property
        value=".topbar-logo"
      />
       to display the logo and make it clickable.
    </li>
  </ul>
  <p>
    With this you get a topbar with the PayEx logo. In the topbar you can add additional functionality like a menu containing links. To enable this add a 
    <PrismCode
      className="language-html"
      component="code"
    >
      &lt;button&gt;
    </PrismCode>
     and a 
    <PrismCode
      className="language-html"
      component="code"
    >
      &lt;nav&gt;
    </PrismCode>
     containing anchors. Remember to include 
    <Attribute
      data={true}
      name="toggle-nav"
      value="{your_nav_id}"
    />
     as a 
    <PrismCode
      className="language-html"
      component="code"
    >
      &lt;button&gt;
    </PrismCode>
     attribute to get the menu to appear when the menu-button is clicked.
  </p>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <Topbar
      id="demo-topbar"
      logout={true}
      topbarContent={
        Object {
          "btn": Object {
            "icon": "menu",
            "text": "Menu",
          },
          "items": Array [
            "Link 1",
            "Link 2",
            "Link 3",
          ],
        }
      }
    />
  </ComponentPreview>
</Fragment>
`;

exports[`Components: Topbar renders 1`] = `
<DocContainer
  docToc={true}
>
  <p
    className="lead"
  >
    The topbar is used to give users an easily available navigational bar at the top of your web application.
  </p>
  <Overview />
  <FixedTopbar />
</DocContainer>
`;
